azurerm/resource_arm_app_configuration.go

```diff
@@ -0,0 +1,385 @@
+
+	return resourceArmAppConfigurationRead(d, meta)
+}
+
+func resourceArmAppConfigurationUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).AppConfiguration.AppConfigurationsClient
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*ArmClient).StopContext, d)
+	defer cancel()
+
+	log.Printf("[INFO] preparing arguments for Azure ARM App Configuration update.")
+
+	name := d.Get("name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+
+	t := d.Get("tags").(map[string]interface{})
+	skuName := d.Get("sku").(string)
+	sku := appconf.Sku{
+		Name: &skuName,
+	}
+
+	parameters := appconf.ConfigurationStoreUpdateParameters{
+		Sku:  &sku,
+		Tags: tags.Expand(t),
+	}
+
+	future, err := client.Update(ctx, resourceGroup, name, parameters)
+	if err != nil {
+		return fmt.Errorf("Error updating App Configuration %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("Error waiting for update of App Configuration %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	read, err := client.Get(ctx, resourceGroup, name)
+	if err != nil {
+		return fmt.Errorf("Error retrieving App Configuration %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+	if read.ID == nil {
+		return fmt.Errorf("Cannot read App Configuration %s (resource Group %q) ID", name, resourceGroup)
+	}
+
+	d.SetId(*read.ID)
+
+	return resourceArmAppConfigurationRead(d, meta)
+}
+
+func resourceArmAppConfigurationRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).AppConfiguration.AppConfigurationsClient
+	ctx, cancel := timeouts.ForRead(meta.(*ArmClient).StopContext, d)
+	defer cancel()
+
+	id, err := azure.ParseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	name := id.Path["configurationStores"]
+
+	resp, err := client.Get(ctx, resourceGroup, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[DEBUG] App Configuration %q was not found in Resource Group %q - removing from state!", name, resourceGroup)
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error making Read request on App Configuration %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	d.Set("name", resp.Name)
+	d.Set("resource_group_name", resourceGroup)
+	if location := resp.Location; location != nil {
+		d.Set("location", azure.NormalizeLocation(*location))
+	}
+
+	if sku := resp.Sku; sku != nil {
+		d.Set("sku", sku.Name)
+	}
+
+	if endpoint := resp.Endpoint; endpoint != nil {
+		d.Set("endpoint", resp.Endpoint)
+	}
+
+	resultPage, err := client.ListKeys(ctx, resourceGroup, name, "")
+	if err != nil {
+		return fmt.Errorf("Failed to receive access keys for App Configuration %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	values := resultPage.Values()
+	for _, value := range values {
+		if &value == nil {
```

Comment:
Unfortunately, this isn't the right check for nil. `value` is actually fine in this case as it's just an APIKey but the individual attributes in an APIKey are pointers `*`. Those need to be nil checked before pointing to them